### DISCUSSION TITLE
Can I use the generated dominant colors for seasonal tone classification (cool/warm)

### DISCUSSION THREAD

**Discussion by soniacarole** [ChenglongMa/SkinToneClassifier#11] [General]
I want to build recommendation clothes matching based on skin tone analysis. how can I build from the generated .csv files since they have 2 dominant colors?

**Comment by ChenglongMa** [comment]
Hi @soniacarole,

Thank you for your attention.

How many dominant colors do you need? You can use the `n_colors` parameter to specify it. For example:

```bash
stone -i /path/to/images/ --n_colors 1
```
Then, there will be only 1 dominant color in the `result.csv`.

BTW, there is a `percent n` column (e.g., `percent 1` and `percent 2`) in the `result.csv`. You can use these values to refine the recommendations.

Please let me know if I can help you.
Thanks!

**Reply by soniacarole** [reply]
Thank you so much for your answers. Your codes came to me as a hope I
would appreciate it if you help me with my work
I would like to build a model for personal color analysis based on skin
tone detection for outfit matching recommendation
let me know if you see an incorrect process in my idea and help me with
some better idea to do it
1- using public face datasets like SCUT-500, LWF
2 detect skin tone color of each skin tone  and classify (cool winter or
warm summer)
for this step basically, there is not a dataset  for color chart I was
thinking of GAn to generate the color matching based on the dominant colors
4  using the users' preferences the detected skin color analysis to
generate outfit  matching(top - bottom-shoes)
Could you help me to achieve it  with any details and steps

**Comment by ChenglongMa** [comment]
Hi @soniacarole,

It is an interesting project.

There are a few issues that I think need attention:

1. Do you have a ground-truth dataset that shows outfits that suits a specific skin tone?
2. Outfit matching recommendation is different from other recommendation systems. It cannot rely solely on the user's personal preferences. More often, we need the knowledge of experts, e.g., [The 4-Season Color Theory System](https://www.prettyyourworld.com/4-season-color-theory.html).
3. In addition to skin color, I think there are many other characteristics that influence user preferences, e.g., demographic characteristics (gender, stature, etc), contextual information (season, location, etc).
4. You can try to search other literature or related work to see how they solve the problem.
    1. [A Complexion based Outfit color recommender using Neural Networks](https://doi.org/10.1109/ICAECT49130.2021.9392418)
    2. [Hierarchical Fashion Graph Network for Personalized Outfit Recommendation](https://doi.org/10.1145/3397271.3401080)
    3. [POG: Personalized Outfit Generation for Fashion Recommendation at Alibaba iFashion](https://doi.org/10.1145/3292500.3330652)

Please let me know if you have any other questions.
Thanks!
